rdflib/paths.py

```diff
@@ -223,20 +224,25 @@ def _n3(
 
 
 @total_ordering
-class Path:
+class Path(ABC):
     __or__: Callable[["Path", Union["URIRef", "Path"]], "AlternativePath"]
     __invert__: Callable[["Path"], "InvPath"]
     __neg__: Callable[["Path"], "NegatedPath"]
     __truediv__: Callable[["Path", Union["URIRef", "Path"]], "SequencePath"]
     __mul__: Callable[["Path", str], "MulPath"]
 
+    @abstractmethod
     def eval(
         self,
         graph: "Graph",
         subj: Optional["_SubjectType"] = None,
         obj: Optional["_ObjectType"] = None,
     ) -> Iterator[Tuple["_SubjectType", "_ObjectType"]]:
-        raise NotImplementedError()
+        pass
+    
+    @abstractmethod
+    def n3(self, namespace_manager: Optional["NamespaceManager"] = None) -> str:
+        pass
```

Comment:
This is mostly a stylistic preference, and really has no practical consequence, but the Python docs suggest using Ellipses (i.e. `...`) here.

```suggestion
        ...
```